### ISSUE TITLE
Append additional key:value to single file style.

### ISSUE THREAD

**Issue by tomgenoni** [sparkartgroup/gulp-markdown-to-json#9]
Thanks for the script. This is great. Had one question: in addition to the front-matter content I wanted to pass in the path to the file — `file.path` — so that with each entry in the resulting json there'd be a "path":"/path/to/file.md". Is that possible?

**Comment by pushred** [comment]
Apologies for the super belated response. For the one year anniversary (!) of your question I added an optional transform function that passes the [Vinyl][vinyl] object which provides this property. This approach (vs. config) has been super helpful for me in other plugins where I've had similar  needs.

To add it to the JSON output:

```js
gulp.src('./content/**/*.md')
  .pipe(gutil.buffer())
  .pipe(markdownToJSON(marked, (data, file) => {
    data.path = file.path;
    return data;
  }))
  .pipe(gulp.dest('.'))
```

[vinyl]: https://github.com/gulpjs/vinyl/tree/v0.4.6

**Comment by pushred** [comment]
This was published as 1.0.0, closing for now, feel free to reopen if you've got any feedback
